src/core/Akka/IO/SelectionHandler.cs

```diff
@@ -262,14 +260,24 @@ private void EnableInterest(SocketChannel channel, SocketAsyncOperation op)
             {
                 switch (op)
                 {
-                        case SocketAsyncOperation.Accept:
-                        case SocketAsyncOperation.Receive:
-                            Execute(() => _read.Add(channel.Socket, channel));                            
-                            break;
-                        case SocketAsyncOperation.Connect:
-                        case SocketAsyncOperation.Send:
-                            Execute(() => _write.Add(channel.Socket, channel));
-                            break;
+                    case SocketAsyncOperation.Accept:
+                    case SocketAsyncOperation.Receive:
+                        Execute(() =>
+                        {
+                            _read.Add(channel.Socket, channel);
```

Comment:
this looks racy to me, _read is a dictionary and is now executed on another thread.
Or?